### PR TITLE
DOC-4696 - add Gerrit to contribution guide 

### PR DESCRIPTION
**Pull request by jamiltz**
https://issues.couchbase.com/browse/DOC-4696

This PR adds the [Gerrit wiki](https://hub.internal.couchbase.com/confluence/display/CR/Contributing+Changes+via+Gerrit) to the contribution guide.

Preview: https://deploy-preview-83--cb-docs-staging.netlify.com/home/contribute/send-pr.html#send-changes-for-review

### CODE REVIEW

**Comment by jamiltz** [comment]
requesting @brett19 and @eric-schneider's review

File: home/modules/contribute/pages/send-pr.adoc [hunk 1]
```diff
@@ -25,14 +25,101 @@ After you complete your changes and fixed any warnings and errors reported by An
...
 . On the Open a Pull Request screen, update the PR title if it needs clarification, and then click btn:[Create Pull Request].
 
+=== Submit a Gerrit Change Request
+
+Some projects use the Gerrit code review process.
+The following sections describe the steps to initialize a repository with Gerrit and how to submit a change request.
+
+==== Initial Account Setup
+
+To submit your first documentation contribution you must sign the contributor license agreement.
```
Comment (amarantha-k) [inline home/modules/contribute/pages/send-pr.adoc]: Reword to something like, "Before you submit your first contribution, you must sign..."

File: home/modules/contribute/pages/send-pr.adoc [hunk 2]
```diff
@@ -25,14 +25,101 @@ After you complete your changes and fixed any warnings and errors reported by An
...
+
+Run through the following steps to submit a change on the Gerrit server.
+
+. Create a new branch where *BRANCH_NAME* is the remote branch to commit changes to.
++
+[source,console]
+----
+git branch bug_666 gerrit/[BRANCH_NAME]
+----
+. Checkout on the topic branch.
```
Comment (amarantha-k) [inline home/modules/contribute/pages/send-pr.adoc]: Switch to the topic branch using `git checkout`

File: home/modules/contribute/pages/send-pr.adoc (hunk 1 above)
Comment (amarantha-k) [inline home/modules/contribute/pages/send-pr.adoc]: Reword to something like, "Before you submit your first contribution, you must sign..."

File: home/modules/contribute/pages/send-pr.adoc [hunk 3]
```diff
@@ -25,14 +25,101 @@ After you complete your changes and fixed any warnings and errors reported by An
...
+----
+git checkout bug_666
+----
+. Push the changes to the Gerrit server.
++
+[source,console]
+----
+git push gerrit HEAD:refs/for/[BRANCH_NAME]
+----
+If successful, you'll see a URL in the command output.
```
Comment (amarantha-k) [inline home/modules/contribute/pages/send-pr.adoc]: Can we add an example URL?

File: home/modules/contribute/pages/send-pr.adoc [hunk 4]
```diff
@@ -25,14 +25,101 @@ After you complete your changes and fixed any warnings and errors reported by An
...
+. Open that URL to access the Gerrit web page and add reviewers.
+Adding a reviewer causes the change to show up in the reviewer's "Reviewable by me" list, and they also get notified by email.
+
 == Next Step
```
Comment (amarantha-k) [inline home/modules/contribute/pages/send-pr.adoc]: Reword to something like, "Before you submit your first contribution, you must sign..."

**Comment by simon-dew** [comment]
Do the sections on *Initial Account Setup* and *Initial Project Setup* belong on this page? They would be better placed on the *Repository Organization* page or the *Create a Working Branch* page. Or perhaps even a page for their own.

**Comment by simon-dew** [comment]
With repositories that have been initialized with git-gerrit, the process of submitting changes to gerrit is different: {{$ git gerrit update -b [BRANCH]}} followed by {{$ git gerrit submit -b [BRANCH]}}.

**Comment by simon-dew** [comment]
Also, for some projects at least, when you submit a gerrit change the default git text editor opens. You have to edit the commit message as appropriate, save the file, and exit the editor to submit the change. Maybe add this step in the **Code Workflow** section, after submitting the changes.

File: home/modules/contribute/pages/send-pr.adoc (hunk 2 above)
Comment (jamiltz) [reply home/modules/contribute/pages/send-pr.adoc]: https://github.com/couchbase/docs-site/pull/83/commits/c90b0d7b931ea1bb29fa94e0f22d914863fd4d0e

File: home/modules/contribute/pages/send-pr.adoc (hunk 1 above)
Comment (jamiltz) [reply home/modules/contribute/pages/send-pr.adoc]: https://github.com/couchbase/docs-site/pull/83/commits/c90b0d7b931ea1bb29fa94e0f22d914863fd4d0e

File: home/modules/contribute/pages/send-pr.adoc (hunk 4 above)
Comment (jamiltz) [reply home/modules/contribute/pages/send-pr.adoc]: https://github.com/couchbase/docs-site/pull/83/commits/c90b0d7b931ea1bb29fa94e0f22d914863fd4d0e

File: home/modules/contribute/pages/send-pr.adoc (hunk 3 above)
Comment (jamiltz) [reply home/modules/contribute/pages/send-pr.adoc]: https://github.com/couchbase/docs-site/pull/83/commits/c90b0d7b931ea1bb29fa94e0f22d914863fd4d0e

**Comment by simon-dew** [comment]
Instructions for contributing to Analytics are [here](https://github.com/couchbase/cbas-core/#contributing-to-couchbase-analytics).

**Comment by jamiltz** [comment]
> Do the sections on Initial Account Setup and Initial Project Setup belong on this page? They would be better placed on the Repository Organization page or the Create a Working Branch page. Or perhaps even a page for their own.

Yes, I think a standalone page for GitHub and one for Gerrit would help. I'll make that change.

> The process for Initial Project Setup doesn't seem to be the same for all repositories. For CBAS, we use git-gerrit, which simplifies the process of initializing the repository and submitting changes. Should maybe add a note saying something like, "this process may vary from one repository to another".

I had no idea about those other workflows. I'll add a note.

**Comment by jamiltz** [comment]
Updated instructions worked for @RichardSmedley 
But it took an extra command for the `commit-msg` hook to work (see [console output](https://gist.github.com/jamiltz/3effba344c48721cf857154e427633f9)). The guide might need updating to include this step if it happens for other contributors.

**Comment by amarantha-k** [comment]
Closing this PR without merge as it needs more work and a new owner to update the content.

**Comment by mojavelinux** [comment]
@amarantha-k Should we still keep this branch or can it be removed?

**Pull request closed without merging**